tests/components/escea/test_config_flow.py

```diff
@@ -0,0 +1,80 @@
+"""Tests for Escea."""
+
+from unittest.mock import Mock, patch
+
+import pytest
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.escea.const import DISPATCH_CONTROLLER_DISCOVERED, ESCEA
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+
+@pytest.fixture
+def mock_disco():
+    """Mock discovery service."""
+    disco = Mock()
+    disco.pi_disco = Mock()
+    disco.pi_disco.controllers = {}
+    yield disco
+
+
+def _mock_start_discovery(hass, mock_disco):
+    def do_disovered(*args):
+        async_dispatcher_send(hass, DISPATCH_CONTROLLER_DISCOVERED, True)
+        return mock_disco
+
+    return do_disovered
+
+
+async def test_not_found(hass, mock_disco):
+    """Test not finding Escea controller."""
+
+    with patch(
+        "homeassistant.components.escea.config_flow.async_start_discovery_service"
```

Comment:
Patch the library `pescea_discovery_service` name in the discovery module instead to get access to the callbacks on the `Listener` instance. Then set the appropriate callback as side effect on `discovery_service.start_discovery` to trigger the test behavior.

Reply:
Hi Martin, thanks for this suggestion... though I'm sorry but I'm new to python and I can't figure out how I could set this up or use it to test the callbacks.

Reply:
I fixed the tests according to my comment. I also renamed the domain constant according to our standards.

Reply:
Thank you so much, and sorry it is taking up your time. I am learning a lot from your review comments, so thanks.